pkg/descheduler/node/node_test.go

```diff
@@ -215,238 +245,527 @@ func TestPodFitsAnyOtherNode(t *testing.T) {
 		pod         *v1.Pod
 		nodes       []*v1.Node
 		success     bool
+		podsOnNodes []*v1.Pod
 	}{
 		{
 			description: "Pod fits another node matching node affinity",
-			pod:         createPodManifest(nodeNames[2], nodeLabelKey, nodeLabelValue),
+			pod: test.BuildTestPod("p1", 0, 0, nodeNames[2], func(pod *v1.Pod) {
+				pod.Spec.NodeSelector = map[string]string{
+					nodeLabelKey: nodeLabelValue,
+				}
+			}),
 			nodes: []*v1.Node{
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						Name: nodeNames[0],
-						Labels: map[string]string{
-							nodeLabelKey: nodeLabelValue,
+				test.BuildTestNode(nodeNames[0], 64000, 128*1000*1000*1000, 200, func(node *v1.Node) {
+					node.ObjectMeta.Labels = map[string]string{
+						nodeLabelKey: nodeLabelValue,
+					}
+
+					node.Status.Allocatable[v1.ResourceEphemeralStorage] = *resource.NewQuantity(1000*1000*1000*1000, resource.DecimalSI)
+				}),
+				test.BuildTestNode(nodeNames[1], 64000, 128*1000*1000*1000, 200, func(node *v1.Node) {
+					node.ObjectMeta.Labels = map[string]string{
+						nodeLabelKey: "no",
+					}
+
+					node.Status.Allocatable[v1.ResourceEphemeralStorage] = *resource.NewQuantity(1000*1000*1000*1000, resource.DecimalSI)
+				}),
+				test.BuildTestNode(nodeNames[2], 0, 0, 0, nil),
+			},
+			podsOnNodes: []*v1.Pod{},
+			success:     true,
+		},
+		{
+			description: "Pod expected to fit one of the nodes",
```

Comment:
How does this test differs from `Pod fits another node matching node affinity`? I only see node labels of `node1` and `node2` switched.